### ISSUE TITLE
Expanding to long multi line text temporary freezes x11

### ISSUE THREAD

**Issue by Zulgrib** [autokey/autokey#242]
## Classification:

Performance

## Reproducibility:

Always

## Version

AutoKey version: 0.95.1

Used GUI (Gtk, Qt, or both): Gtk

If the problem is known to be present in more than one version, please list all of those.

Installed via: PPA

Linux Distribution: Mint19

## Summary

Expanding to long multi lines text feeeze the whole interface few seconds

## Steps to Reproduce (if applicable)

- Make a long multi line text (did 22 lines with text)
- Expand it

## Expected Results

- Text appear instantly without locking me from doing something else

## Actual Results

- Instead, everything x11 related is frozen until the text appears, time required seems to depend on the mono thread performances of the machine

## Notes

Paste using keyboard

**Comment by luziferius** [comment]
I just tested it, and the same happens on my machine.
In the past, you were able to watch AutoKey type the phrase. The freeze seems to be caused by some (more-or-less recent) X server update.
But even when it worked without freezing the X server, it took roughly the same time to expand the phrase using the keyboard.
Unfortunately, the keyboard typing mode was never really instant.

Regarding the expected results: It won’t be possible to do something else, while autokey types. 
You can use the clipboard mode, which is way faster for large phrases.

(Side note: If you wish to use the clipboard mode, you can grab the packages for the newest version from the [GitHub releases page](https://github.com/autokey/autokey/releases). Version 0.95.4 contains some fixes for bugs affecting the clipboard pasting mode. Upcoming version 0.95.5 will contain some additional fixes.)

**Comment by Zulgrib** [comment]
If it's not something that could be improved, when using "dumb" text file, could there be a warning if a too-long-text is detected ?

Why not updating the PPA to 0.95.4 ?

**Comment by josephj11** [comment]
@Zulgrib Yes, please do the PPA update! JK, but Thomas only has so many hours to work on things and that just hasn't made it to the top of his list for awhile. On occasion we have had more than one active developer at a time, but it's been pretty rare.

**Comment by Zulgrib** [comment]
I see, as this problem is not critical and the workaround works fantastically well, I will stay at curent PPA version out of lazyness.

**Comment by luziferius** [comment]
The PPA has a currently inactive maintainer (@sporkwitch). I don’t have write access there.

That’s one of the reasons I attach self-built and self-signed Debian packages to each release.

Building and uploading to a PPA seems to be not that time consuming. The Wiki has a page about [packaging AutoKey](https://github.com/autokey/autokey/wiki/Packaging). The local package building takes about a minute. But creating yet another PPA will only cause more confusion and duplicates.
